File: yx_model/src/main/java/com/yx/model/Sku/Sku.java
package com.yx.model.Sku;

import com.yx.model.Base.comm;
import lombok.Getter;
import lombok.Setter;

public class Sku {

    //主键
    @Getter @Setter
    private String skuId;

    //商品所属客户
    @Getter @Setter
    private String customerId;

    //集团ID
    @Getter @Setter
    private String groupId;

    //条形码,可多个，用分号（;）隔开
    @Getter @Setter
    private String skuNo;

    //商品名称
    @Getter @Setter
    private String skuName;

    //商品类型 (ZC=正常商品,FX=分销商品,ZH=组合商品,ZP=赠品,BC=包材,HC=耗材,FL=辅料,XN=虚拟品,FS=附属品,CC=残次品,OTHER=其它)
    @Getter @Setter
    private String skuType = "ZC";

    //货号
    @Getter @Setter
    private String goodsNo = skuNo;

    //供应商名称
    @Getter @Setter
    private String supplierId = "";

    //商品简称
    @Getter @Setter
    private String shortName = skuName;

    //英文名
    @Getter @Setter
    private String englishName = "";

    //商品属性 (如红色, XXL)
    @Getter @Setter
    private String skuProperty = "";

    //商品计量单位
    @Getter @Setter
    private String stockUnit = "";

    //长 (厘米)
    @Getter @Setter
    private double length;

    //宽 (厘米)
    @Getter @Setter
    private double width;

    //高 (厘米)
    @Getter @Setter
    private double height;

    //体积 (升)
    @Getter @Setter
    private double volume;

    //毛重 (千克)
    @Getter @Setter
    private double grossWeight;

    //净重 (千克)
    @Getter @Setter
    private double netWeight;

    //颜色
    @Getter @Setter
    private String color = "";

    //尺寸
    @Getter @Setter
    private String size = "";

    //渠道中的商品标题
    @Getter @Setter
    private String title = "";

    //商品类别ID
    @Getter @Setter
    private String categoryId = "";

    //商品类别名称
    @Getter @Setter
    private String categoryName = "";

    //计价货类
    @Getter @Setter
    private String pricingCategory = "";

    //安全库存
    @Getter @Setter
    private int safetyStock;

    //吊牌价
    @Getter @Setter
    private double tagPrice;

    //零售价
    @Getter @Setter
    private double retailPrice;

    //成本价
    @Getter @Setter
    private double costPrice;

    //采购价
    @Getter @Setter
    private double purchasePrice;

    //季节编码
    @Getter @Setter
    private String seasonCode = "";

    //季节名称
    @Getter @Setter
    private String seasonName = "";

    //品牌编号
    @Getter @Setter
    private String brandId = "";

    //是否需要串号管理, Y/N (默认为N)
    @Getter @Setter
    private String isSNMgmt = "N";

    //是否需要保质期管理, Y/N (默认为N)
    @Getter @Setter
    private String isShelfLifeMgmt = "N";

    //保质期 (小时)
    @Getter @Setter
    private int shelfLife;

    //保质期禁收天数
    @Getter @Setter
    private int rejectLifecycle;

    //保质期禁售天数
    @Getter @Setter
    private int lockupLifecycle;

    //保质期临期预警天数
    @Getter @Setter
    private int adventLifecycle;

    //是否需要批次管理, Y/N (默认为N)
    @Getter @Setter
    private String isBatchMgmt = "N";

    //批次代码
    @Getter @Setter
    private String batchCode = "";

    //包装代码
    @Getter @Setter
    private String packCode = "";

    //箱规
    @Getter @Setter
    private String pcs = "";

    //商品的原产地
    @Getter @Setter
    private String originAddress = "";

    //批准文号
    @Getter @Setter
    private String approvalNumber = "";

    //是否易碎品, Y/N,  (默认为N)
    @Getter @Setter
    private String isFragile = "N";

    //是否危险品, Y/N,  (默认为N)
    @Getter @Setter
    private String isHazardous = "N";

    //商品包装材料类型
    @Getter @Setter
    private String packageMaterial = "";

    //备注
    @Getter @Setter
    private String remark = "";

    //创建时间,YYYY-MM-DD HH:MM:SS
    @Getter @Setter
    private String createTime;

    //更新时间,YYYY-MM-DD HH:MM:SS
    @Getter @Setter
    private String updateTime;

    //是否有效, Y/N (默认为Y)
    @Getter @Setter
    private String validate = "Y";

    //扩展字段


    //客户
    @Getter @Setter
    private String customerName;

    //供应商
    @Getter @Setter
    private String supplierName;

    //规格
    private String spec;

    //规格
    public String getSpec() { return length+"*"+width+"*"+height+"="+volume; }

    //商品类型描述
    private String skuTypeName;

    //商品类型描述
    public String getSkuTypeName() {
        return comm.SkuType.get(skuType);
    }
}

File: yx_wms_web/src/main/resources/static/sku/sku.js
/* 商品管理 */

let app;
$(document).ready(function () {
    app = new Vue({
        //定义页面作用域
        el: "#app",
        //定义页面变量
        data: {
            search: { p_text: { text: "skuNo", value: "" }, customerId: "", isShelfLifeMgmt: ""},
            checkAll: true,
            tag: {
                brandList: [],
                skuTypeList: [],
                customerList: [],
                seasonTypeList: [],
                dataWhetherList: []
            },
            modalPage: {
                title: "添加商品",
                show: false,
                width: "100px",
                model: {},
                default: {
                    skuType: "ZC",
                    isShelfLifeMgmt: "N",
                    isBatchMgmt: "N",
                    isSNMgmt: "N"
                },
                //表单绑定验证
                validate: {
                    customerId: [
                        {
                            validator: (rule, value, callback) => {
                                if (!lutils.validate.required(value)) {
                                    callback(new Error("请输入数据"));
                                } else {
                                    callback();
                                }
                            }, trigger: "blur"
                        }
                    ],
                    skuNo: [
                        {
                            validator: (rule, value, callback) => {
                                if (!lutils.validate.required(value)) {
                                    callback(new Error("请输入数据"));
                                } else {
                                    callback();
                                }
                            }, trigger: "blur"
                        }
                    ],
                    skuName: [
                        {
                            validator: (rule, value, callback) => {
                                if (!lutils.validate.required(value)) {
                                    callback(new Error("请输入数据"));
                                } else {
                                    callback();
                                }
                            }, trigger: "blur"
                        }
                    ]
                }
            },
            modalValidate: {
                title: "",
                show: false,
                table: {
                    validate: "",
                    height: "60%",
                    columns: []
                }
            },
            selectRows: []
        },
        {
            //页面初始化
            init(); {
                setTimeout(function () {
                    $(window).resize();
                }, 100);
                //省市区（数据源）
                $.getJSON("http://staticyx.lnetco.com/cons/region.json", function (result) {
                    app.tag.regions = result || [];
                });
                //获取数据（是·否）列表
                $.ajax({
                    url: "/SelectList/GetDataWhetherSelectList",
                    success: function (result) {
                        app.tag.dataWhetherList = result;
                    }
                });
                //获取商品类型列表
                $.ajax({
                    url: "/SelectList/GetSkuTypeSelectList",
                    success: function (result) {
                        app.tag.skuTypeList = result;
                    }
                });
                //获取用户拥有的客户公司列表
                $.ajax({
                    url: "/SelectList/GetUserCustomerSelectList",
                    success: function (result) {
                        app.tag.customerList = result;
                    }
                });
                //获取季节列表
                $.ajax({
                    url: "/SelectList/GetSeasonTypeSelectList",
                    success: function (result) {
                        app.tag.seasonTypeList = result;
                    }
                });
            },
            //查询数据
            onSearch(); {
                this.options.dataSource.filter(YX.Kendo.sformat(app.search));
            },
            //编辑商品信息
            editSku(skuId); {
                app.modalPage.show = true;
                app.modalPage.model = app.modalPage.default;

                //重置表单验证警告信息
                app.$nextTick(function() {
                    //$nextTick方法为解决第一次执行clearValidate报错
                    app.$refs.modalPage.clearValidate();
                });
                if(!skuId){
                    app.modalPage.title = "添加商品信息";
                    return;
                }
                //加载商品信息
                $.ajax({
                    url: "/Sku/GetSkuInfo?skuId="+skuId,
                    success: function (result) {
                        app.modalPage.model = result;
                    }
                });
                app.modalPage.title = "编辑商品信息";
            },
            changeCustomerSelect(); {
                //获取客户的供应商列表
                $.ajax({
                    url: "/SelectList/GetCustomerSupplierSelectList?customerId="+app.modalPage.model.customerId,
                    success: function (result) {
                        app.tag.supplierList = result;
                    }
                });
            },
            //保存商品信息
            saveStock(); {
                //验证数据完整性和正确性
                app.$refs.modalPage.validate((valid) => {
                    if(;!valid;) return false;

                    //保存数据
                    $.ajax({
                        url: "/Sku/SaveSku",
                        data: JSON.stringify(app.modalPage.model),
                        dataType: "text", //返回值类型（默认是JSON）
                        success: function (result) {
                            if (result > 0) {
                                app.modalPage.show = false;//关闭弾罩层
                                YX.Alert.success("操作成功！");
                                app.onSearch(); //重新查询
                            } else {
                                YX.Alert.error("操作数据失败，请联系管理员！");
                            }
                        }
                    });
            })
            },
            //弹出启用/删除框
            changeDataValidate(title); {
                this.selectRows = YX.Kendo.getGridSelectRows(app.grid);
                if (this.selectRows.length <= 0) {
                    YX.Alert.warning("请选择一条数据！");
                    return;
                }
                this.modalValidate.title = "确定要" + title + "以下数据吗?";
                switch (title) {
                    case "启用":
                        this.modalValidate.table.validate = "Y";
                        break;
                    case "删除":
                        this.modalValidate.table.validate = "603";
                        break;
                }
                this.modalValidate.show = true;
            },
            //保存数据状态
            updateDataValidate(); {
                let skuId = this.selectRows[0].skuId;
                let validate = this.modalValidate.table.validate;
                $.ajax({
                    url: "/Sku/UpdateDataValidate?skuId="+ skuId +"&validate="+ validate,
                    dataType: "text",//返回值类型（默认是JSON）
                    success: function (result) {
                        if (result > 0) {
                            YX.Alert.success("操作成功！");
                            app.onSearch();//重新查询
                        } else {
                            YX.Alert.error("数据操作失败！");//错误信息弹出
                        }
                        app.modalValidate.show = false;//关闭弹出页面
                    }
                });
            }
        }
})
    //数据列表初始化
    app.grid = $("#grid").kendoGrid(app.options = YX.Kendo.getGridConf({
        //设置数据列
        columns: [
            {
                field: "rownumber",
                title: " ",
                template: (dataItem) => app.options.dataSource.data().indexOf(dataItem) + 1,
                width;: 30
            },
            {
                "商品条码",
                field;: "skuNo",
                width;: 100
            },
            {
                "商品名称",
                template;: "<a title='修改' onclick=\"app.editSku('#= skuId#')\" >#= skuName# </a>",
                width;: 200
            },
            {
                "商品类型",
                field;: "skuTypeName",
                width;: 80
            },
            {
                "所属客户",
                field;: "customerName",
                width;: 100
            },
            {
                "供应商",
                field;: "supplierName",
                width;: 100
            },
            {
                "规格",
                field;: "spec",
                width;: 100
            },
            {
                "颜色",
                field;: "color",
                width;: 100
            },
            {
                "尺寸",
                field;: "size",
                width;: 100
            },
            {
                "分类",
                field;: "categoryName",
                width;: 100
            },
            {
                "保质期管理",
                field;: "isShelfLifeMgmt",
                width;: 100
            },
            {
                "保质期(H)",
                field;: "shelfLife",
                width;: 100
            },
            {
                "批次管理",
                field;: "isBatchMgmt",
                width;: 100
            },
            {
                "SN管理",
                field;: "isSNMgmt",
                width;: 100
            },
            {
                "安全库存数",
                field;: "safetyStock",
                width;: 80
            },
            {
                "吊牌价",
                field;: "tagPrice",
                width;: 50
            },
            {
                "零售价",
                field;: "retailPrice",
                width;: 50
            },
            {
                "成本价",
                field;: "costPrice",
                width;: 50
            },
            {
                "采购价",
                field;: "purchasePrice",
                width;: 50
            },
        ],
        //设置数据源
        YX.Kendo.getGridSrc("listId", "/Sku/GetSkuList")
}))
    //页面初始化
    app.init();
    //===延时渲染===
    setTimeout(() => {
        window.onresize = () => { YX.Kendo.gridResize() };
        $(".main_loading").fadeOut(500);
        YX.Kendo.gridResize();
},
    200;
)
});